aws/resource_aws_elasticache_cluster.go

```diff
@@ -425,85 +427,68 @@ func resourceAwsElasticacheClusterRead(d *schema.ResourceData, meta interface{})
 	conn := meta.(*AWSClient).elasticacheconn
 	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
 
-	req := &elasticache.DescribeCacheClustersInput{
-		CacheClusterId:    aws.String(d.Id()),
-		ShowCacheNodeInfo: aws.Bool(true),
+	c, err := finder.CacheClusterWithNodeInfoByID(conn, d.Id())
+	if tfresource.NotFound(err) {
+		log.Printf("[WARN] ElastiCache Cache Cluster (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
 	}
-
-	res, err := conn.DescribeCacheClusters(req)
 	if err != nil {
-		if isAWSErr(err, elasticache.ErrCodeCacheClusterNotFoundFault, "") {
-			log.Printf("[WARN] ElastiCache Cluster (%s) not found", d.Id())
-			d.SetId("")
-			return nil
+		return fmt.Errorf("error reading ElastiCache Cache Cluster (%s): %w", d.Id(), err)
+	}
+
+	d.Set("cluster_id", c.CacheClusterId)
+	d.Set("node_type", c.CacheNodeType)
+	d.Set("num_cache_nodes", c.NumCacheNodes)
+	d.Set("engine", c.Engine)
+	d.Set("engine_version", c.EngineVersion)
+	if c.ConfigurationEndpoint != nil {
+		d.Set("port", c.ConfigurationEndpoint.Port)
+		d.Set("configuration_endpoint", aws.String(fmt.Sprintf("%s:%d", *c.ConfigurationEndpoint.Address, *c.ConfigurationEndpoint.Port)))
+		d.Set("cluster_address", aws.String((*c.ConfigurationEndpoint.Address)))
```

Comment:
Dereference safety and extraneous wrapping:

```suggestion
		d.Set("cluster_address", c.ConfigurationEndpoint.Address)
```

Reply:
Yikes, that was a copy-paste I didn't check